src/port1.0/portinstall.tcl

```diff
@@ -308,6 +308,19 @@ proc portinstall::create_archive {location archive.type} {
     puts $fd "@cxx_stdlib_overridden ${portinstall::cxx_stdlib_overridden}"
     close $fd
 
+    if {[get_statefile_value "SOURCE_DATE_EPOCH" $target_state_fd source_date_epoch] != 0 && $reproducible_timestamps == 1} {
+        # Set standardised file modification times
+        fs-traverse file $destpath {
+            if {[file type $file] == "link"} {
+                # file mtime doesn't work on symbolic links
+                exec touch -ht $source_date_epoch $file
```

Comment:
I would rather see the MacPorts [`touch` procedure](https://github.com/macports/macports-base/blob/82b064968d0ae5561911b850249939a7f77ae158/src/port1.0/portutil.tcl#L1168) be used than using the external `touch` command. The MacPorts `touch` procedure doesn't support `-h` yet but perhaps that could be added.